### PR TITLE
[IOTDB-902] Optimize max_time aggregation and fix order by time desc aggregation bug 

### PR DESCRIPTION
**Pull request by Alima777**
This pr:
1. Fix some bugs of `order by time desc`.
2. Add more tests of `order by time desc`.
3. Optimize the max_time and last_value aggregation process without value filter using descReader

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationAlignByDeviceIT.java [hunk 1]
```diff
@@ -1,113 +0,0 @@
...
-import java.sql.Statement;
-import org.apache.iotdb.db.utils.EnvironmentUtils;
-import org.apache.iotdb.jdbc.Config;
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Test;
-
-public class IoTDBAggregationAlignByDeviceIT {
-
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationAlignByDeviceIT.java:37]: I don't get the reason why you remove this IT?

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java [hunk 2]
```diff
@@ -357,11 +438,11 @@ private void minMaxTimeAggreWithSingleFilterTest() throws ClassNotFoundException
...
-                  "," + resultSet.getString(min_time(d0s3)) +
-                  "," + resultSet.getString(min_time(d0s4));
+              resultSet.getString(min_time(d0s0)) + "," +
+              resultSet.getString(min_time(d0s1)) + "," +
+              resultSet.getString(min_time(d0s2)) +
+              "," + resultSet.getString(min_time(d0s3)) +
+              "," + resultSet.getString(min_time(d0s4));
           Assert.assertEquals(ans, retArray[cnt]);
           cnt++;
         }
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java:448]: Better to use try with resource reduce code smell...

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java [hunk 3]
```diff
@@ -370,27 +451,70 @@ private void minMaxTimeAggreWithSingleFilterTest() throws ClassNotFoundException
...
+                  .getString(max_time(d0s2))
+                  + "," + resultSet.getString(max_time(d0s3)) + "," + resultSet
+                  .getString(max_time(d0s4));
+          Assert.assertEquals(ans, retArray[cnt]);
+          cnt++;
+        }
+        Assert.assertEquals(2, cnt);
+      } finally {
+        resultSet.close();
+      }
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java:473]:
```suggestion
      try (resultSet = statement.getResultSet()) {
        while (resultSet.next()) {
          String ans =
              resultSet.getString(TIMESTAMP_STR) + "," + resultSet.getString(max_time(d0s0))
                  + "," + resultSet.getString(max_time(d0s1)) + "," + resultSet
                  .getString(max_time(d0s2))
                  + "," + resultSet.getString(max_time(d0s3)) + "," + resultSet
                  .getString(max_time(d0s4));
          Assert.assertEquals(ans, retArray[cnt]);
          cnt++;
        }
        Assert.assertEquals(2, cnt);
      }
```

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java [hunk 4]
```diff
@@ -370,27 +451,70 @@ private void minMaxTimeAggreWithSingleFilterTest() throws ClassNotFoundException
...
+              resultSet.getString(min_time(d0s2)) +
+              "," + resultSet.getString(min_time(d0s3)) +
+              "," + resultSet.getString(min_time(d0s4));
           Assert.assertEquals(ans, retArray[cnt]);
           cnt++;
         }
         Assert.assertEquals(1, cnt);
       } finally {
         resultSet.close();
       }
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java:496]: Same as above

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java [hunk 5]
```diff
@@ -370,27 +451,70 @@ private void minMaxTimeAggreWithSingleFilterTest() throws ClassNotFoundException
...
+                  .getString(max_time(d0s2))
+                  + "," + resultSet.getString(max_time(d0s3)) + "," + resultSet
+                  .getString(max_time(d0s4));
+          Assert.assertEquals(ans, retArray[cnt]);
+          cnt++;
+        }
+        Assert.assertEquals(2, cnt);
+      } finally {
+        resultSet.close();
+      }
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java:517]: Same...

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java [hunk 6]
```diff
@@ -416,12 +540,35 @@ private void minValueAggreWithSingleFilterTest() throws ClassNotFoundException,
...
+                    .getString(min_value(d0s2))
+                    + "," + resultSet.getString(min_value(d0s3)) + "," + resultSet
+                    .getString(min_value(d0s4));
+            Assert.assertEquals(ans, retArray[cnt]);
+            cnt++;
+          }
+          Assert.assertEquals(1, cnt);
+        } finally {
+          resultSet.close();
+        }
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationLargeDataIT.java:554]: Same...
There are a lot of `try finally`, better to fix them together.

File: server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java [hunk 7]
```diff
@@ -36,9 +36,9 @@
   @Test
   public void avgAggrResultTest() throws QueryProcessException, IOException {
     AggregateResult avgAggrResult1 = AggregateResultFactory
-        .getAggrResultByName(SQLConstant.AVG, TSDataType.DOUBLE);
+        .getAggrResultByName(SQLConstant.AVG, TSDataType.DOUBLE,true);
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java:39]:
```suggestion
        .getAggrResultByName(SQLConstant.AVG, TSDataType.DOUBLE, true);
```

File: server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java [hunk 8]
```diff
@@ -36,9 +36,9 @@
...
     AggregateResult avgAggrResult2 = AggregateResultFactory
-        .getAggrResultByName(SQLConstant.AVG, TSDataType.DOUBLE);
+        .getAggrResultByName(SQLConstant.AVG, TSDataType.DOUBLE,true);
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java:41]:
```suggestion
        .getAggrResultByName(SQLConstant.AVG, TSDataType.DOUBLE, true);
```

File: server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java [hunk 9]
```diff
@@ -62,9 +62,9 @@ public void avgAggrResultTest() throws QueryProcessException, IOException {
   @Test
   public void maxValueAggrResultTest() throws QueryProcessException, IOException {
     AggregateResult maxValueAggrResult1 = AggregateResultFactory
-        .getAggrResultByName(SQLConstant.MAX_VALUE, TSDataType.DOUBLE);
+        .getAggrResultByName(SQLConstant.MAX_VALUE, TSDataType.DOUBLE,true);
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java:65]:
```suggestion
        .getAggrResultByName(SQLConstant.MAX_VALUE, TSDataType.DOUBLE, true);
```

File: server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java [hunk 10]
```diff
@@ -62,9 +62,9 @@ public void avgAggrResultTest() throws QueryProcessException, IOException {
...
     AggregateResult maxValueAggrResult2 = AggregateResultFactory
-        .getAggrResultByName(SQLConstant.MAX_VALUE, TSDataType.DOUBLE);
+        .getAggrResultByName(SQLConstant.MAX_VALUE, TSDataType.DOUBLE,true);
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java:67]:
```suggestion
        .getAggrResultByName(SQLConstant.MAX_VALUE, TSDataType.DOUBLE, true);
```

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/GroupByFilter.java [hunk 11]
```diff
@@ -65,7 +65,7 @@ public boolean satisfyStartEndTime(long startTime, long endTime) {
       return false;
     else if (startTime <= this.startTime)
       return true;
```
Comment (HTHou) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/GroupByFilter.java:67]: Change to `<` here?

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationAlignByDeviceIT.java (hunk 1 above)
Comment (Alima777) [reply server/src/test/java/org/apache/iotdb/db/integration/IoTDBAggregationAlignByDeviceIT.java:37]: As all the `align by device` test is included in `IoTDBAlignByDeviceIT`. It's redundant here.

File: server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java (hunk 7 above)
Comment (Alima777) [reply server/src/test/java/org/apache/iotdb/db/query/aggregation/AggregateResultTest.java:39]: Fixed.

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/GroupByFilter.java (hunk 11 above)
Comment (Alima777) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/GroupByFilter.java:67]: I think it's <= here. @JackieTien97

Comment (HTHou) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/GroupByFilter.java:67]: Yes, you're right.

**Comment by HTHou** [comment]
There are some conflicts, please fix them...

**Review by HTHou — APPROVED** [review]

**Pull request merged**